[Bug]: 部分Agent上报信息报错

### Preflight Checklist

- [X] I agree to follow the [Code of Conduct](https://github.com/HXSecurity/DongTai/blob/main/.github/CODE_OF_CONDUCT.md) that this project adheres to.
- [X] I have searched the [issue tracker](https://www.github.com/HXSecurity/DongTai/issues) for an issue that matches the one I want to file, without success.
- [X] I am not looking for support or already pursued the available [support channels](https://github.com/HXSecurity/DongTai/issues/new/choose) without success.

### Version

1.8.5

### Installation Type

Official Kubernetes

### Service Name

DongTai-agent-java

### Describe the details of the bug and the steps to reproduce it

如题，部分Agent的上报事件报错，导致上报信息不完整。

**报错案例1**



`2022-09-28 11:40:00 [io.dongtai.iast.agent] [ERROR] nulljava.lang.NullPointerException
        at io.dongtai.iast.core.handler.hookpoint.graphy.GraphBuilder.convertToReport(GraphBuilder.java:104)
        at io.dongtai.iast.core.handler.hookpoint.graphy.GraphBuilder.buildAndReport(GraphBuilder.java:29)
        at io.dongtai.iast.core.handler.hookpoint.SpyDispatcherImpl.leaveDubbo(SpyDispatcherImpl.java:156)
        at org.apache.dubbo.monitor.support.MonitorFilter.invoke(MonitorFilter.java:89)
        at org.apache.dubbo.rpc.protocol.ProtocolFilterWrapper$1.invoke(ProtocolFilterWrapper.java:83)
        at org.apache.dubbo.rpc.protocol.dubbo.filter.FutureFilter.invoke(FutureFilter.java:51)
        at org.apache.dubbo.rpc.protocol.ProtocolFilterWrapper$1.invoke(ProtocolFilterWrapper.java:83)
        at org.apache.dubbo.rpc.filter.ConsumerContextFilter.invoke(ConsumerContextFilter.java:69)
        at org.apache.dubbo.rpc.protocol.ProtocolFilterWrapper$1.invoke(ProtocolFilterWrapper.java:83)
        at org.apache.dubbo.rpc.protocol.InvokerWrapper.invoke(InvokerWrapper.java:56)
        at org.apache.dubbo.rpc.cluster.support.FailoverClusterInvoker.doInvoke(FailoverClusterInvoker.java:82)
        at org.apache.dubbo.rpc.cluster.support.AbstractClusterInvoker.invoke(AbstractClusterInvoker.java:260)
        at org.apache.dubbo.rpc.cluster.interceptor.ClusterInterceptor.intercept(ClusterInterceptor.java:47)
        at org.apache.dubbo.rpc.cluster.support.wrapper.AbstractCluster$InterceptorInvokerNode.invoke(AbstractCluster.java:92)
        at org.apache.dubbo.rpc.cluster.support.wrapper.MockClusterInvoker.invoke(MockClusterInvoker.java:88)
        at org.apache.dubbo.rpc.proxy.InvokerInvocationHandler.invoke(InvokerInvocationHandler.java:74)
        at org.apache.dubbo.common.bytecode.proxy17.getPersonalCampName(proxy17.java)
        at cn.com.x x x.personal.integration.RpcPersonalCampServiceClient.getPersonalCampName(RpcPersonalCampServiceClient.java:56)
        at cn.com.xxx.wechat.bo.wxmsgsender.XPerPersonalLinkingWxMsgSender.sendPasswordToApp(XPersonalLinkingWxMsgSender.java:237)
        at cn.com.xxx.wechat.bo.wxmsgsender.XPerPersonalLinkingWxMsgSender.check(XPerPersonalLinkingWxMsgSender.java:115)
        at cn.com.xxx.wechat.bo.wxmsgsender.WxMsgSenderChecker.check(WxMsgSenderChecker.java:24)
        at cn.com.xxx.core.job.CrontableUpgradeJobHandler.sendTrainerTemplateMsg(CrontableUpgradeJobHandler.java:334)
        at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.lang.reflect.Method.invoke(Method.java:498)
        at com.xxl.job.core.handler.impl.MethodJobHandler.execute(MethodJobHandler.java:29)
        at com.xxl.job.core.thread.JobThread.run(JobThread.java:165)`

**报错案例2**



`2022-09-28 11:40:00 [io.dongtai.iast.agent] [ERROR] nulljava.lang.NullPointerException
        at io.dongtai.iast.core.handler.hookpoint.graphy.GraphBuilder.convertToReport(GraphBuilder.java:104)
        at io.dongtai.iast.core.handler.hookpoint.graphy.GraphBuilder.buildAndReport(GraphBuilder.java:29)
        at io.dongtai.iast.core.handler.hookpoint.SpyDispatcherImpl.leaveDubbo(SpyDispatcherImpl.java:156)
        at org.apache.dubbo.monitor.support.MonitorFilter.invoke(MonitorFilter.java:89)
        at org.apache.dubbo.rpc.protocol.ProtocolFilterWrapper$1.invoke(ProtocolFilterWrapper.java:83)
        at org.apache.dubbo.rpc.protocol.dubbo.filter.FutureFilter.invoke(FutureFilter.java:51)
        at org.apache.dubbo.rpc.protocol.ProtocolFilterWrapper$1.invoke(ProtocolFilterWrapper.java:83)
        at org.apache.dubbo.rpc.filter.ConsumerContextFilter.invoke(ConsumerContextFilter.java:69)
        at org.apache.dubbo.rpc.protocol.ProtocolFilterWrapper$1.invoke(ProtocolFilterWrapper.java:83)
        at org.apache.dubbo.rpc.protocol.InvokerWrapper.invoke(InvokerWrapper.java:56)
        at org.apache.dubbo.rpc.cluster.support.FailoverClusterInvoker.doInvoke(FailoverClusterInvoker.java:82)
        at org.apache.dubbo.rpc.cluster.support.AbstractClusterInvoker.invoke(AbstractClusterInvoker.java:260)
        at org.apache.dubbo.rpc.cluster.interceptor.ClusterInterceptor.intercept(ClusterInterceptor.java:47)
        at org.apache.dubbo.rpc.cluster.support.wrapper.AbstractCluster$InterceptorInvokerNode.invoke(AbstractCluster.java:92)
        at org.apache.dubbo.rpc.cluster.support.wrapper.MockClusterInvoker.invoke(MockClusterInvoker.java:88)
        at org.apache.dubbo.rpc.proxy.InvokerInvocationHandler.invoke(InvokerInvocationHandler.java:74)
        at org.apache.dubbo.common.bytecode.proxy17.getPersonalCampName(proxy17.java)
        at cn.com.xxx.personal.integration.RpcPersonalCampServiceClient.getPersonalCampName(RpcPersonalCampServiceClient.java:56)
        at cn.com.xxx.wechat.bo.wxmsgsender.XPerPersonalLinkingWxMsgSender.checkSendPasskitLinkingMsg(XPerPersonalLinkingWxMsgSender.java:251)
        at cn.com.xxx.wechat.bo.wxmsgsender.XPerPersonalLinkingWxMsgSender.check(XPerPersonalLinkingWxMsgSender.java:117)
        at cn.com.xxx.wechat.bo.wxmsgsender.WxMsgSenderChecker.check(WxMsgSenderChecker.java:24)
        at cn.com.xxx.core.job.CrontableUpgradeJobHandler.sendTrainerTemplateMsg(CrontableUpgradeJobHandler.java:334)
        at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.lang.reflect.Method.invoke(Method.java:498)
        at com.xxl.job.core.handler.impl.MethodJobHandler.execute(MethodJobHandler.java:29)
        at com.xxl.job.core.thread.JobThread.run(JobThread.java:165)`

### Additional Information

_No response_

### Logs

_No response_

--- a/dongtai-core/src/main/java/io/dongtai/iast/core/handler/hookpoint/controller/impl/SourceImpl.java
+++ b/dongtai-core/src/main/java/io/dongtai/iast/core/handler/hookpoint/controller/impl/SourceImpl.java
@@ -161,6 +161,9 @@ public class SourceImpl {
      */
     public static Set<Object> parseCustomModel(Object model) {
         Set<Object> modelValues = new HashSet<Object>();
+        if (!TaintPoolUtils.isNotEmpty(model)) {
+            return modelValues;
+        }
         Class<?> sourceClass = model.getClass();
         if (sourceClass.getClassLoader() == null) {
             return modelValues;

--- a/dongtai-core/src/main/java/io/dongtai/iast/core/handler/hookpoint/framework/dubbo/DubboHandler.java
+++ b/dongtai-core/src/main/java/io/dongtai/iast/core/handler/hookpoint/framework/dubbo/DubboHandler.java
@@ -45,20 +45,18 @@ public class DubboHandler {
         // todo: register server
         if (attachments != null) {
             Map<String, String> requestHeaders = new HashMap<String, String>(attachments.size());
-            for (Map.Entry<String, String> entry : attachments.entrySet()) {
-                requestHeaders.put(entry.getKey(), entry.getValue());
-            }
+            requestHeaders.putAll(attachments);
             if (null == EngineManager.SERVER) {
                 // todo: read server addr and send to OpenAPI Service
                 EngineManager.SERVER = new IastServer(requestHeaders.get("dubbo"), 0, "TCP",true);
             }
             Map<String, Object> requestMeta = new HashMap<String, Object>(12);
-            requestMeta.put("protocol", "dubbo/" + requestHeaders.get("dubbo"));
+            requestMeta.put("protocol", "dubbo/" + (requestHeaders.get("dubbo") == null ? requestHeaders.get("dubbo") : "unknown"));
             requestMeta.put("scheme", "dubbo");
             requestMeta.put("method", "RPC");
             requestMeta.put("secure", "true");
             requestMeta.put("requestURL", dubboService.split("\\?")[0]);
-            requestMeta.put("requestURI", requestHeaders.get("path"));
+            requestMeta.put("requestURI", requestHeaders.get("path") == null ? requestHeaders.get("path") : "");
             requestMeta.put("remoteAddr", "");
             requestMeta.put("queryString", "");
             requestMeta.put("headers", requestHeaders);
@@ -191,37 +189,48 @@ public class DubboHandler {
             if (TaintPoolUtils.isJdkType(arg)) {
                 modelItems.add(arg);
             } else {
-                modelItems.add(SourceImpl.parseCustomModel(arg));
+                Set<Object> models = SourceImpl.parseCustomModel(arg);
+                if (models.size() > 0) {
+                    modelItems.add(models);
+                }
             }
         }
         boolean isHitTaints = false;
         for (Object item : modelItems) {
             isHitTaints = isHitTaints || TaintPoolUtils.poolContains(item, event);
         }
-        if (isHitTaints) {
-            int invokeId = SpyDispatcherImpl.INVOKE_ID_SEQUENCER.getAndIncrement();
-            event.setInvokeId(invokeId);
-            event.setPlugin("DUBBO");
-            // todo: 获取 service name
-            event.setServiceName("");
-            // todo: 获取 traceId
-            event.setTraceId(sharedTraceId.get());
-            event.setCallStack(StackUtils.getLatestStack(5));
-            EngineManager.TRACK_MAP.addTrackMethod(invokeId, event);
-            Set<Object> resModelItems = SourceImpl.parseCustomModel(rpcResult);
-            Set<Object> resModelSet = new HashSet<Object>();
-            for (Object obj : resModelItems) {
-                // fixme: 暂时只跟踪字符串相关内容
-                if (obj instanceof String) {
-                    resModelSet.add(obj);
-                    int identityHashCode = System.identityHashCode(obj);
-                    event.addTargetHash(identityHashCode);
-                    event.addTargetHashForRpc(obj.hashCode());
-                    EngineManager.TAINT_HASH_CODES.add(identityHashCode);
-                }
-            }
-            event.setOutValue(resModelSet);
+        if (!isHitTaints) {
+            return;
         }
+
+        int invokeId = SpyDispatcherImpl.INVOKE_ID_SEQUENCER.getAndIncrement();
+        event.setInvokeId(invokeId);
+        event.setPlugin("DUBBO");
+        // todo: 获取 service name
+        event.setServiceName("");
+        // todo: 获取 traceId
+        event.setTraceId(sharedTraceId.get());
+        event.setCallStack(StackUtils.getLatestStack(5));
+        Set<Object> resModelItems = SourceImpl.parseCustomModel(rpcResult);
+        Set<Object> resModelSet = new HashSet<Object>();
+        for (Object obj : resModelItems) {
+            // fixme: 暂时只跟踪字符串相关内容
+            if (obj instanceof String) {
+                if (((String) obj).isEmpty()) {
+                    continue;
+                }
+                resModelSet.add(obj);
+                int identityHashCode = System.identityHashCode(obj);
+                event.addTargetHash(identityHashCode);
+                event.addTargetHashForRpc(obj.hashCode());
+                EngineManager.TAINT_HASH_CODES.add(identityHashCode);
+            }
+        }
+        if (resModelSet.isEmpty()) {
+            return;
+        }
+        event.setOutValue(resModelSet);
+        EngineManager.TRACK_MAP.addTrackMethod(invokeId, event);
     }
 
     public static void solveServiceExit(Object invocation, Object rpcResult) {

